pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -6,173 +6,171 @@ package zedmanager
-		base.OldVolumeConfigLogType, key).Infof("App instance to volume relation.")
+		base.VolumeRefConfigLogType, vrs.Key()).Infof("App instance to volume relation.")
 }
 
-func lookupVolumeConfig(ctx *zedmanagerContext, key string) *types.OldVolumeConfig {
+func lookupVolumeRefConfig(ctx *zedmanagerContext, key string) *types.VolumeRefConfig {
 
-	pub := ctx.pubVolumeConfig
+	pub := ctx.pubVolumeRefConfig
 	c, _ := pub.Get(key)
 	if c == nil {
-		log.Infof("lookupVolumeConfig(%s) not found", key)
+		log.Infof("lookupVolumeRefConfig(%s) not found", key)
 		return nil
 	}
-	config := c.(types.OldVolumeConfig)
+	config := c.(types.VolumeRefConfig)
 	return &config
 }
 
-// Note that this function returns the entry even if Pending* is set.
-func lookupVolumeStatus(ctx *zedmanagerContext, blobSha256 string,
-	appInstID uuid.UUID, volumeID uuid.UUID, purgeCounter uint32) *types.OldVolumeStatus {
+func lookupVolumeRefStatus(status *types.AppInstanceStatus,
+	vrc types.VolumeRefConfig) *types.VolumeRefStatus {
 
-	key := types.VolumeKeyFromParts(blobSha256, appInstID, volumeID, purgeCounter)
-	sub := ctx.subVolumeStatus
-	st, _ := sub.Get(key)
-	if st == nil {
-		log.Infof("lookupVolumeStatus(%s) not found", key)
-		return nil
+	log.Debugf("lookupVolumeRefStatus(%v)", vrc.VolumeID)
+	for i := range status.VolumeRefStatusList {
+		vrs := &status.VolumeRefStatusList[i]
+		if vrs.VolumeID == vrc.VolumeID && vrs.GenerationCounter == vrc.GenerationCounter {
+			log.Debugf("lookupVolumeRefStatus(%v) found %s generationCounter %d",
+				vrs.VolumeID, vrs.DisplayName, vrs.GenerationCounter)
+			return vrs
+		}
 	}
-	status := st.(types.OldVolumeStatus)
-	return &status
+	log.Debugf("lookupVolumeRefStatus(%v) Done", vrc.VolumeID)
+	return nil
 }
 
-func publishVolumeConfig(ctx *zedmanagerContext,
-	status *types.OldVolumeConfig) {
+func publishVolumeRefConfig(ctx *zedmanagerContext, config *types.VolumeRefConfig) {
 
-	key := status.Key()
-	log.Infof("publishVolumeConfig(%s)", key)
-	pub := ctx.pubVolumeConfig
-	pub.Publish(key, *status)
+	key := config.Key()
+	log.Debugf("publishVolumeRefConfig(%s)", key)
+	pub := ctx.pubVolumeRefConfig
+	pub.Publish(key, *config)
+	log.Debugf("publishVolumeRefConfig(%s) Done", key)
 }
 
-func unpublishVolumeConfig(ctx *zedmanagerContext, uuidStr string) {
+func unpublishVolumeRefConfig(ctx *zedmanagerContext, key string) {
 
-	key := uuidStr
-	log.Infof("unpublishVolumeConfig(%s)", key)
-	pub := ctx.pubVolumeConfig
+	log.Debugf("unpublishVolumeRefConfig(%s)", key)
+	pub := ctx.pubVolumeRefConfig
 	c, _ := pub.Get(key)
 	if c == nil {
-		log.Errorf("unpublishVolumeConfig(%s) not found", key)
+		log.Errorf("unpublishVolumeRefConfig(%s) not found", key)
 		return
 	}
 	pub.Unpublish(key)
+	log.Debugf("unpublishVolumeRefConfig(%s) Done", key)
+}
+
+func lookupVolumeStatus(ctx *zedmanagerContext, vrs *types.VolumeRefStatus) *types.VolumeStatus {
```

Comment:
When/why do you look for volumeStatus? I thought you'd just operate on VolumeRefStatus.

Reply:
You might be able to delete this function and subVolumeStatus?